src/Tvheadend.cpp

```diff
@@ -2158,6 +2165,23 @@ bool CTvheadend::ParseEvent ( htsmsg_t *msg, bool bAdd, Event &evt )
   /* Recordings complete */
   SyncDvrCompleted();
 
+  if (htsmsg_get_s64(msg, "stop", &stop) && bAdd)
+  {
+    Logger::Log(LogLevel::LEVEL_ERROR, "malformed eventAdd: 'stop' missing");
+    return false;
+  }
+
+  // tvheadend's "epgMaxTime" implementation is buggy in htsp versions prior to 25.
+  // To workaround this, we must request all epg data and filter it here.
+  if (m_conn.GetProtocol() < 25 &&
+      Settings::GetInstance().GetAsyncEpg() &&
+      m_epgMaxDays > EPG_TIMEFRAME_UNLIMITED)
+  {
+    // check event's stop time against m_epgMaxDays, return if to large
+    if (stop > static_cast<int64_t>(time(NULL) + m_epgMaxDays * int64_t(24 * 60 *60)))
+      return false;
+  }
+
```

Comment:
Not parsing the event causes the same problem as the buggy "epgMaxTime". An event will be sent once in async mode, if we ignore an event, it will be missing forever (without restarting the htsp connction). It's better to just ignore the day setting for the affected servers or would this lead in an unlimited tv guide on the Kodi side?